src/templates/course.js

```diff
@@ -39,8 +39,18 @@ const Program = ({data, pageContext, yml}) => {
     setOpen(false);
   };
 
-  const apply_button_text = session && session.location ? session.location.button.apply_button_text : "Apply";
-  const syllabus_button_text = session && session.location ? session.location.button.syllabus_button_text : "Download Syllabus";
+  const [applyButtonText, setApplyButtonText] = useState("");
+  let city = session && session.location ? session.location.city : [];
+  let currentLocation = data.allLocationYaml.edges.find(loc => loc.node?.city === city);
+
+  useEffect( () => {
+    if (currentLocation !== undefined){
+      setApplyButtonText(currentLocation.node.button.apply_button_text)
+    }
+  }, [currentLocation])
+
+  const apply_button_text = session && session.location ? applyButtonText : "Apply";
```

Comment:
This is an unnecessary lines of code, you should use the `applyButtonText` state instead of the ~~`apply_button_text`~~ variable